### PR TITLE
[WFLY-14871] Logic in try/finally blocks hides exceptions (IIOP)

### PR DESCRIPTION
**Pull request by boris-unckel**
Fixes https://issues.redhat.com/browse/WFLY-14871

### CODE REVIEW

**Comment by bstansberry** [comment]
@tadamski Please review

File: iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/naming/jndi/CNCtx.java [hunk 1]
```diff
@@ -406,27 +408,52 @@ private String getStringifiedIor(String url) throws NamingException {
...
+                            suppressed = ne;
+                        }
                     }
-                } catch (IOException e) {
-                    NamingException ne = IIOPLogger.ROOT_LOGGER.invalidURLOrIOR(url);
-                    ne.setRootCause(e);
-                    throw ne;
                 }
+                if (suppressed != null) {
+                    if (suppressed instanceof RuntimeException) {
```
Comment (tadamski) [inline iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/naming/jndi/CNCtx.java:441]: Maybe we could simplify it a bit and just do or and rethrow. Why do we need the cast?

**Review by tadamski — COMMENTED** [review]
The logic seems good to me. I'm wondering whether the code should be simplified a little bit as described in a comment above.

File: iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/naming/jndi/CNCtx.java (hunk 1 above)
Comment (boris-unckel) [reply iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/naming/jndi/CNCtx.java:441]: @tadamski in general this "pattern" is the way I solved all similar cases. In new Line 420 the Throwable may contain the RuntimeException or Error. You can't rethrow a undeclared Throwable, therefor the cast.

Comment (boris-unckel) [reply iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/naming/jndi/CNCtx.java:441]: @tadamski Christian Beikov had the same concern at the same point. He refactored it this way: [https://github.com/hibernate/hibernate-orm/blob/wip/6.0/hibernate-core/src/main/java/org/hibernate/internal/util/ExceptionHelper.java](https://github.com/hibernate/hibernate-orm/blob/wip/6.0/hibernate-core/src/main/java/org/hibernate/internal/util/ExceptionHelper.java) It works, but it completely undermines the typesafety of throw / Throwables. I don't want to provide it without broader review/acceptance.

**Review by tadamski — APPROVED** [review]

**Pull request merged**